rflx/pyrflx/typevalue.py

```diff
@@ -515,17 +515,28 @@ def _next_field(self, fld: str) -> str:
     def _prev_field(self, fld: str) -> str:
         if fld == INITIAL.name:
             return ""
+        prev: List[str] = []
         for l in self._type.incoming(Field(fld)):
             if self.__simplified(l.condition) == TRUE:
-                return l.source.name
+                prev.append(l.source.name)
+
+        if len(prev) == 1:
+            return prev[0]
+        for field in prev:
+            if self._fields[field].set:
+                return field
```

Comment:
Couldn't you just replace this code by:

```Python
for l in self._type.incoming(Field(fld)):
    if self.__simplified(l.condition) == TRUE and self._fields[l.source.name].set:
        return l.source.name
```

Or in other words: Just adding `and self._fields[l.source.name].set` to the previous version?

Reply:
That would change the behaviour of `preset_fields` and subsequently the `accessible_fields` function in a way that breaks 13 unit tests in their current form. Currently `accessible_fields` returns all fields that could be set next because their length and first are known. After this change all fields have to be set in the right sequence. (not sure what the expected behaviour is here because normally all fields would be set in the right sequence anyway)

I wrote it that way because it is just an issue if there is more than one predecessor with a valid edge condition.

Reply:
I see. I think it is preferable to be able to set fields in arbitrary sequence whenever possible. But doesn't your current solution also change the behavior of `accessible_fields`? A field with two predecessors is only accessible now if one of its predecessors is set while this wasn't necessarily the case before.

Reply:
That's right. I changed `prev_field` to decide among all possible predecessors with no edge condition by checking which one is accessible instead of set. I don't think that a situation in which there is more than one predecessor field with no edge condition accessible can occur, but i am not quite certain.